src/node_buffer.cc

```diff
@@ -19,7 +19,11 @@
   } while (0)
 
 #define ARGS_THIS(argT)                                                     \
-  Local<Object> obj = argT;                                                 \
+  Handle<Value> _obj = argT;                                                \
+  if (!HasInstance(_obj)) {                                                 \
+    return env->ThrowTypeError("must be called on Buffer instance");        \
```

Comment:
> Since `class` won't be out until 2 (since we're increasing the major, i guess...).

That is actually about now. Class-in-strict will be live in 2.0 due to 4.2, iirc.

> Are we going to require this be a 3 release?

Well, v8 4.3 is already going to breaking enough to warrant a 3.0, right? :/